### PR TITLE
add parity-scale-codec

### PR DESCRIPTION
**Pull request by alvicsam**

### CODE REVIEW

**Review by sergejparity — APPROVED** [review]

File: dockerfiles/parity-scale-codec/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+ARG BUILD_DATE
+ARG REGISTRY_PATH=docker.io/paritytech
+
+FROM ${REGISTRY_PATH}/base-ci-linux:latest
+
+# metadata
+LABEL io.parity.image.authors="devops-team@parity.io" \
+	io.parity.image.vendor="Parity Technologies" \
+	io.parity.image.title="${REGISTRY_PATH}/ink-ci-linux" \
+	io.parity.image.description="Inherits from docker.io/paritytech/base-ci. \
```
Comment (TriplEight) [inline dockerfiles/parity-scale-codec/Dockerfile:11]: base-ci-linux:latest

File: dockerfiles/parity-scale-codec/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+	rust nightly, clippy, rustfmt, miri, rust-src grcov, rust-covfix, llvm-tools-preview, \
+	cargo-contract, xargo, binaryen, parallel, codecov" \
+	io.parity.image.source="https://github.com/paritytech/scripts/blob/${VCS_REF}/\
+	dockerfiles/ink-ci-linux/Dockerfile" \
```
Comment (TriplEight) [inline dockerfiles/parity-scale-codec/Dockerfile:15]: parity-scale-codec

File: dockerfiles/parity-scale-codec/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+	io.parity.image.documentation="https://github.com/paritytech/scripts/blob/${VCS_REF}/\
+	dockerfiles/ink-ci-linux/README.md" \
```
Comment (TriplEight) [inline dockerfiles/parity-scale-codec/Dockerfile:17]: parity-scale-codec

File: dockerfiles/parity-scale-codec/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+
+FROM ${REGISTRY_PATH}/base-ci-linux:latest
+
+# metadata
+LABEL io.parity.image.authors="devops-team@parity.io" \
+	io.parity.image.vendor="Parity Technologies" \
+	io.parity.image.title="${REGISTRY_PATH}/ink-ci-linux" \
+	io.parity.image.description="Inherits from docker.io/paritytech/base-ci. \
+	rust nightly, clippy, rustfmt, miri, rust-src grcov, rust-covfix, llvm-tools-preview, \
+	cargo-contract, xargo, binaryen, parallel, codecov" \
```
Comment (TriplEight) [inline dockerfiles/parity-scale-codec/Dockerfile:13]: gnupg

File: dockerfiles/parity-scale-codec/README.md [hunk 5]
```diff
@@ -0,0 +1,58 @@
+# parity-scale-codec CI for Linux Distributionsimage
```
Comment (TriplEight) [inline dockerfiles/parity-scale-codec/README.md:1]: _

File: dockerfiles/parity-scale-codec/README.md [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+Docker image based on our base CI image `<base-ci-linux:latest>`.
+
+Used to build and test parity-scale-codec.
+
+## Dependencies and Tools
+
+- `parallel`: for running commands in parallel, without overlapping output
+- `codecov`: to upload the test coverage results
+- `binaryen`: needed by cargo-contract for optimizing Wasm files
+
```
Comment (TriplEight) [inline dockerfiles/parity-scale-codec/README.md:12]: gnupg

**Review by TriplEight — COMMENTED** [review]
was this a temp image?

**Pull request merged**